consumer/consumererror/consumererror.go

```diff
@@ -0,0 +1,165 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"google.golang.org/grpc/status"
+)
+
+type Error struct {
```

Comment:
I think is actually better to make multiple simple errors that we join, this is the golang new pattern.

Reply:
I've given this a shot. It feels a little strange to me since I expect `errors.Join` to be used with errors that are produced independently, e.g. validating a slice of data points and joining errors produced from validation failures in separate lines.

Our errors are produced from a single underlying error, e.g. an HTTP status of 500. So you either wrap the same error multiple times if the simple errors can be used independently, or you require that they're joined with the underlying error using `errors.Join`. In the second case it feels strange to me that they implement `error` when they're not really errors but just metadata around another error. What do you think?

Reply:
@bogdandrutu Could you take a look when you have an opportunity? Do you think using `errors.Join` is an improvement?

Reply:
@bogdandrutu, using the `errors.Join` to represent one error would conflict with the fanout behavior, where we do the same to combine several errors from different exporters. If we do this, it'll be hard to differentiate between them. WDYT?